src/v/archival/ntp_archiver_service.h

```diff
@@ -562,6 +603,19 @@ class ntp_archiver {
     /// If true, we are holding up trimming of local storage
     bool local_storage_pressure() const;
 
+    // If true, a flush request has been issued and is still in progress.
+    bool flush_in_progress() const;
+
+    // If true, the manifest's last offset is greater than the provided offset,
+    // and the archival stm is clean past the provided offset.
+    bool uploaded_and_clean_past_offset(model::offset o) const;
+
+    // If true, the current flush operation has uploaded past the set
```

Comment:
can you clarify what "current flush operation" means? what if two flushes were called? seems like there is a little ambiguity here?

Reply:
Rewrote comment clarifying that "current" flush operation refers to the "most recently issued" operation.

If `flush()` is called multiple times, users who haven't resolved `wait()` futures may experience undesired behavior, as waiters are only notified when the most recently issued flush has completed.